Classes/Updater.lua

```diff
@@ -13,127 +13,63 @@ cUpdater = {}
 
 local g_CheckAttempts = 0
 local g_DownloadAttempts = 0
-local g_ReceivedPluginInfo = ""
-local g_LatestVersion = nil
-local g_WorldEditZip = nil
 
 
 
 
-
-local VersionCheckForNewerVersionerCallbacks =
-{
-	OnConnected = function(a_TCPLink)
-		local res, msg = a_TCPLink:StartTLSClient()
-		a_TCPLink:Send("GET /cuberite/WorldEdit/master/Info.lua HTTP/1.0\r\nHost: raw.githubusercontent.com\r\n\r\n")
-	end,
-	
-	OnError = function(a_TCPLink, a_ErrorCode, a_ErrorMsg) 
-		LOGWARNING("Error while checking for newer WorldEdit version: " .. a_ErrorMsg)
-	end,
-	
-	OnReceivedData = function(a_TCPLink, a_Data)
-		g_ReceivedPluginInfo = g_ReceivedPluginInfo .. a_Data
-	end,
-	
-	OnRemoteClosed = function(a_TCPLink)
-		cUpdater:ParseResults()
-	end,
-}
-
-
-
-
-
-local NewVersionDownloaderCallbacks =
-{
-	OnConnected = function(a_TCPLink)
-		-- Start a secured connection
-		local res, msg = a_TCPLink:StartTLSClient("", "", "")
-		
-		-- Send Request Header
-		a_TCPLink:Send("GET /cuberite/WorldEdit/zip/master HTTP/1.0\r\nHost: codeload.github.com\r\nConnection: Keep-alive\r\n\r\n")
-		
-		-- Clear data we might have from previous attempts
-		g_WorldEditZip = ""
-	end,
-	
-	OnError = function(a_TCPLink, a_ErrorCode, a_ErrorMsg)
-		if (g_DownloadAttempts < g_Config.Updates.NumAttempts) then
-			g_DownloadAttempts = g_DownloadAttempts + 1
-			cUpdater:DownloadLatestVersion()
-		else
-			LOGWARNING("Error while downloading newer worldedit version: " .. a_ErrorCode)
-		end
-	end,
-	
-	OnReceivedData = function(a_TCPLink, a_Data)
-		-- Append the received Data to the saved data
-		g_WorldEditZip = g_WorldEditZip .. a_Data
-	end,
+function cUpdater:CheckForNewerVersion()
+	if (g_CheckAttempts > g_Config.Updates.NumAttempts) then
+		LOGWARNING("Could not connect to Github to check for a newer version for WorldEdit")
+		return;
+	end
+	g_CheckAttempts = g_CheckAttempts + 1;
```

Comment:
How about storing this in the `cUpdater` itself? Access via `self.m_CheckAttempts`